### ISSUE TITLE
Strang code in __libc_exit

### ISSUE THREAD

**Issue by th-otto** [freemint/mintlib#61]
Not really a big deal, but i noticed that in libc_exit https://github.com/freemint/mintlib/blob/2e1fa5ac71b41054be69d5fb6c05d3f7ba00fa16/mintlib/libc_exit.S#L22-L27 some registers are saved that are not used there. However d3 (which **is** used) is not saved (but that's not a problem either, since the program is about to exit)

Any idea about why this was done?

**Comment by mikrosk** [comment]
"Added ColdFire support to most assembler files (except FPU). Contributed by Vincent Riviere." in  38ded951.

@vinriviere any idea? :)

**Comment by th-otto** [comment]
Vincent only changed the movem, but it was already there before (just with a numeric value for the register list)

**Comment by mikrosk** [comment]
Wait, what? You can specify the registers as an immediate operand? Wow.

Anyway, this looks like some ancient leftover code.

**Comment by th-otto** [comment]
> Wait, what? You can specify the registers as an immediate operand? Wow.

Yes. The compiler actually generates it that way (which makes the code hard to read). Especially because it uses a decimal for it.

**Comment by vinriviere** [comment]
> "Added ColdFire support to most assembler files (except FPU). Contributed by Vincent Riviere." in [38ded95](https://github.com/freemint/mintlib/commit/38ded951e07a0db9a667e50b166b5a972d28bd63).

Oh. Thanks for having spotted that. This is obviously a bug.

In that series of "Added ColdFire support" patches, I took particular care to only add the ugly `#ifdef __mcoldfire__` without changing the 68000 code. This one is an exception, where I also replaced the ugly "movem integer" syntax by the standard register list. But I wrongly converted the integer to register list!

0x3020 is actually `d2-d3/a2`. This is the right, original register list.

The wrong `d5/a4/a5` translates to 0x040c. Those values are completely unrelated. I can't see any sensible reason to that mistake. That's just a bug from mine.

So I've just fixed that regression in commit a86fc7600d0cd932c64676c13255e6976a64924f. Many thanks @mikrosk to have spotted that.

BTW: I agree, the whole register save at the top of `libc_exit.S` is totally useless, as the function will never return. I guess it could be completely removed, as well as the frame pointer. Unless it could be useful for gdb or `__mcleanup`. I would be surprised if so.
